app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -5,8 +5,30 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   skip_after_action :verify_policy_scoped
 
   def cis2
-    @user =
-      User.find_or_create_user_from_cis2_oidc(request.env["omniauth.auth"])
-    sign_in_and_redirect @user, event: :authentication # this will throw if @user is not activated
+    user_info = request.env["omniauth.auth"]
+    raw_info = user_info["extra"]["raw_info"]
+    selected_role = raw_info["selected_roleid"]
+    nrbac_role =
+      raw_info["nhsid_nrbac_roles"].find do
+        _1["person_roleid"] == selected_role
+      end
+
+    team = Team.find_by(ods_code: nrbac_role["org_code"])
+    if team.nil?
+      @selected_org =
+        raw_info["nhsid_user_orgs"].find do
+          _1["org_code"] == nrbac_role["org_code"]
+        end
+
+      flash[:org_name] = @selected_org["org_name"]
+      flash[:org_code] = @selected_org["org_code"]
+      flash[:has_other_roles] = raw_info["nhsid_nrbac_roles"].length > 1
```

Comment:
I guess as long as we don't assign to `flash[:info]` or `:success` or `:warning` this is fine... it shouldn't leak to the UI in the flash banner component and it won't persist past the next request.

Should we pass a hash along instead of 3 separate things?

Reply:
Sure, a hash can be tidier.